node/src/service.rs

```diff
@@ -1063,4 +1315,209 @@ where
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    /// Get a stream of new best block notifications.
+    async fn new_best_notification_stream(
+        &self,
+    ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        self.relay_chain_interface
+            .new_best_notification_stream()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    /// Get a stream of finality notifications.
+    async fn finality_notification_stream(
+        &self,
+    ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        self.relay_chain_interface
+            .finality_notification_stream()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    async fn genesis_data(
+        &self,
+        relay_parent: PHash,
+        para_id: ParaId,
+    ) -> OrchestratorChainResult<Option<ContainerChainGenesisData>> {
+        let res: Option<ContainerChainGenesisData> = call_remote_runtime_function(
+            &self.relay_chain_interface,
+            "RegistrarApi_genesis_data",
+            relay_parent,
+            &para_id,
+        )
+        .await
+        .unwrap();
+
+        Ok(res)
+    }
+
+    async fn boot_nodes(
+        &self,
+        relay_parent: PHash,
+        para_id: ParaId,
+    ) -> OrchestratorChainResult<Vec<Vec<u8>>> {
+        let res: Vec<Vec<u8>> = call_remote_runtime_function(
+            &self.relay_chain_interface,
+            "RegistrarApi_boot_nodes",
+            relay_parent,
+            &para_id,
+        )
+        .await
+        .unwrap();
+
+        Ok(res)
+    }
+
+    async fn latest_block_number(
+        &self,
+        relay_parent: PHash,
+        para_id: ParaId,
+    ) -> OrchestratorChainResult<Option<BlockNumber>> {
+        let res: Option<BlockNumber> = call_remote_runtime_function(
+            &self.relay_chain_interface,
+            "AuthorNotingApi_latest_block_number",
+            relay_parent,
+            &para_id,
+        )
+        .await
+        .unwrap();
+
+        Ok(res)
+    }
+
+    async fn best_block_hash(&self) -> OrchestratorChainResult<PHash> {
+        self.relay_chain_interface
+            .best_block_hash()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    async fn finalized_block_hash(&self) -> OrchestratorChainResult<PHash> {
+        self.relay_chain_interface
+            .finalized_block_hash()
+            .await
+            .map_err(|e| OrchestratorChainError::Application(Box::new(e)))
+    }
+
+    async fn data_preserver_active_assignment(
+        &self,
+        _orchestrator_parent: PHash,
+        _profile_id: DataPreserverProfileId,
+    ) -> OrchestratorChainResult<DataPreserverAssignment<ParaId>> {
+        unimplemented!("Data preserver node does not support Starlight yet")
```

Comment:
If possible, remove this `unimplemented` as well.

Reply:
This will only happen if someone runs a node with `--solochain --data-preserver` flags at the same time, which is not implemented yet, so I think this is the best to handle that case

Reply:
*roadmap*